### ISSUE TITLE
Items added with ncselector_additem don't show their short description

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#846]
See #843. I added some code to the `selector` PoC to exercise `ncselector_additem()` following setup of the `ncselector`. The images there summarize things better than I could here; adding the new item and rerendering doesn't show the short description, though it does show the long description.

@sikfeng helped me find this with his original report. Thanks!

**Comment by dankamongmen** [comment]
![2020-08-01-011636_722x857_scrot](https://user-images.githubusercontent.com/143473/89094744-5a7dc080-d395-11ea-8649-dabd72bc4ea9.png)

**Comment by dankamongmen** [comment]
```
[schwarzgerat](130) $ cat e
Term: 78x104 alacritty (alacritty terminal emulator)
printing option [first] [this is the first option]
printing option [2nd] [this is the second option]
printing option [3] [third, third, third option am i]
printing option [fourth] [i have another option here]
printing option [first] [this is the first option]
printing option [2nd] [this is the second option]
printing option [3] [third, third, third option am i]
printing option [fourth] [i have another option here]
printing option [ten] [stunning and brave]
printing option [first] [this is the first option]
printing option [2nd] [this is the second option]
printing option [3] [third, third, third option am i]
printing option [ten] [stunning and brave]
printing option [blah] [this is my crappy description]
printing option [first] [this is the first option]
printing option [2nd] [this is the second option]
printing option [first] [this is the first option]
printing option [2nd] [this is the second option]
printing option [3] [third, third, third option am i]
printing option [fourth] [i have another option here]
[schwarzgerat](0) $
```

it's getting through there, just not getting displayed, odd...

**Comment by sikfeng** [comment]
I just tried your patch and another problem I noticed is that when the focused item is the lowest on the screen it moves the focused option off the screen and does not scroll properly until I scroll back onto the screen. Is that a issue with the library or just the demo?
![2020-08-01-133327_1366x768_scrot](https://user-images.githubusercontent.com/9458918/89094943-a5caba00-d3fb-11ea-8df0-783f91454401.png)

**Comment by sikfeng** [comment]
Sometimes the short description is written where the long description should be, which can be seen when the description is shorter than the option
![image](https://user-images.githubusercontent.com/9458918/89095714-2214cb80-d403-11ea-8a1d-ca56f803783f.png)

```
      ncselector_item ns_item1 = {"no work", ":("};
      ncselector_item ns_item2 = {"no work", ":(("};
      this->m_channel_selector->additem(&ns_item1); 
      this->m_channel_selector->additem(&ns_item2); 
```

**Comment by dankamongmen** [comment]
I see two issues: we're not seeing up `opcolumns` on addition, and for some cases (not seen here), we're not extending `longop` when we need to. I'll knock these both out.

**Comment by dankamongmen** [comment]
Thanks @sikfeng ; i'll do some extensive testing with this over the weekend. Apparently it's awash in problems, yee-haw

**Comment by dankamongmen** [comment]
![2020-08-01-034805_936x936_scrot](https://user-images.githubusercontent.com/143473/89097100-d1bd4f80-d3a9-11ea-9b75-9e8e5b85b3a9.png)

got that bit wrapped up, committing now

**Comment by dankamongmen** [comment]
@sikfeng using the C API, i was able to add multiple options just fine with the following sample code:

```
[schwarzgerat](0) $ git diff
diff --git src/poc/selector.c src/poc/selector.c
index 9e183ed7..f28b609e 100644
--- src/poc/selector.c
+++ src/poc/selector.c
@@ -33,10 +33,23 @@ run_selector(struct notcurses* nc, struct ncselector* ns){
   notcurses_render(nc);
   char32_t keypress;
   ncinput ni;
+  int iter = 4;
   while((keypress = notcurses_getc_blocking(nc, &ni)) != (char32_t)-1){
     if(!ncselector_offer_input(ns, &ni)){
       switch(keypress){
         case NCKEY_ENTER: ncselector_destroy(ns, NULL); return;
+        case 'a': {
+                    char a[20];
+                    snprintf(a, sizeof(a), "blubber%d", iter++);
+                    struct ncselector_item nitem = {
+                      .option = a,
+                      .desc = "this is my crappy description",
+                    };
+                    if(ncselector_additem(ns, &nitem)){
+                      return;
+                    }
+                  }
+                  break;
         case 'M': case 'J': if(ni.ctrl){ ncselector_destroy(ns, NULL); return; }
       }
       if(keypress == 'q'){
```

![2020-08-01-035412_722x857_scrot](https://user-images.githubusercontent.com/143473/89097182-aab34d80-d3aa-11ea-9e38-cdee1edb054f.png)

does your additional information above pertain to the C++ API? i want to keep that over in #846, just to keep things straight. the C API is working properly for me following commit 7e43077.

**Comment by sikfeng** [comment]
I believe the problem remaining is that the `ncplane` needs to be resized when the added items are too long. In your example, that is not a problem because the selector was already initialized with long descriptions.

**Comment by dankamongmen** [comment]
OK, thanks for the confirmation @sikfeng , I'm closing this as fixed, and if @grendello doesn't get a chance to look at #843 today, I'll dig into it for you. I really appreciate you drawing this attention to `ncselector` -- I also filed and fixed #847 this afternoon. We'll have a much better `ncselector` and `ncmultiselector` in 1.7.0 thanks to you.
